tests/isolation_provider/__init__.py

```diff
@@ -0,0 +1,49 @@
+import subprocess
+import tempfile
+
+import pytest
+
+from dangerzone.document import Document
+
+from .. import BASIC_SAMPLE_PDF, test_docs_dir
+
+SVG = """<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
+<svg width="1" height="1" viewBox="-1 -1 1 1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+<rect fill="#fff" stroke="#000" x="-1" y="-1" width="2" height="2"/>
+</svg>
+"""
+
+
+@pytest.fixture
+def long_pdf() -> str:
+    """Generates a 70000 page PDF"""
+
+    svg_file = tempfile.NamedTemporaryFile(mode="w")
+    pdf_file_one_page = tempfile.NamedTemporaryFile(mode="w")
+    pdf_file_1000_pages = tempfile.NamedTemporaryFile(mode="w")
+    pdf_file_70000_pages = tempfile.NamedTemporaryFile(mode="w")
+
+    svg_file.write(SVG)
+    svg_file.flush()
+
+    # Transform SVG "pixel" into PDF page
+    subprocess.run(["convert", f"svg:{svg_file.name}", f"pdf:{pdf_file_one_page.name}"])
+
+    # Combine into 1000 page PDF
+    subprocess.run(
+        ["pdfunite", *([pdf_file_one_page.name] * 1000), pdf_file_1000_pages.name]
+    )
+
+    # Multiply by 70 to obtain a 70K page PDF (~100mb)
+    subprocess.run(
+        ["pdfunite", *([pdf_file_1000_pages.name] * 70), pdf_file_70000_pages.name]
+    )
+
+    yield pdf_file_70000_pages.name
+
+    # Delete files
+    svg_file.close()
+    pdf_file_one_page.close()
+    pdf_file_1000_pages.close()
+    pdf_file_70000_pages.close()
```

Comment:
This block has two issues:

1. It requires having some specific commands in the user's system. This may not be the case in Windows/MacOS, and even on Linux systems.
2. It takes a bit of a time, and produces a big document.

Instead, let's precompute this document. We can create a document with 70K pages with:

```
gs -sDEVICE=pdfwrite -o sample-70k-pages.pdf -dDEVICEWIDTHPOINTS=612 -dDEVICEHEIGHTPOINTS=792 -c  70000 {showpage} repeat
```

This takes roughly 21MiB of space, and we can compress it down to 1.8MiB.

We can also use the same technique (minus the compression) to create a document with large width/height, since we want to catch errors for that as well.

Reply:
That's exactly what I was looking for. Thank you.

Reply:
closing via 5cc9f17